### PR TITLE
conformance: allow error status in Route

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/kubernetes-sigs/gateway-api/issues/1135

This changes tests to assert their is either no parent at all, or a
non-accepted parent (indicating an error message, presumably).

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: conformance/utils/kubernetes/helpers.go [hunk 1]
```diff
@@ -177,6 +177,41 @@ func WaitForGatewayAddress(t *testing.T, client client.Client, gwName types.Name
 	return net.JoinHostPort(ipAddr, port), waitErr
 }
 
+// HTTPRouteMustHaveParents waits for the specified HTTPRoute to have either no parents
```
Comment (skriss) [inline conformance/utils/kubernetes/helpers.go:180]:
```suggestion
// HTTPRouteMustHaveNoAcceptedParents waits for the specified HTTPRoute to have either no parents
```

**Review by skriss — COMMENTED** [review]
LGTM other than one comment nit

File: conformance/utils/kubernetes/helpers.go [hunk 2]
```diff
@@ -177,6 +177,41 @@ func WaitForGatewayAddress(t *testing.T, client client.Client, gwName types.Name
...
+// or a single parent that is not acceptted. This is used to validate HTTPRoute errors.
```
Comment (robscott) [inline conformance/utils/kubernetes/helpers.go:181]:
```suggestion
// or a single parent that is not accepted. This is used to validate HTTPRoute errors.
```

File: conformance/utils/kubernetes/helpers.go [hunk 3]
```diff
@@ -177,6 +177,41 @@ func WaitForGatewayAddress(t *testing.T, client client.Client, gwName types.Name
...
+func HTTPRouteMustHaveNoAcceptedParents(t *testing.T, client client.Client, routeName types.NamespacedName, seconds int) {
+	t.Helper()
+
+	var actual []v1alpha2.RouteParentStatus
+	waitFor := time.Duration(seconds) * time.Second
+	waitErr := wait.PollImmediate(1*time.Second, waitFor, func() (bool, error) {
```
Comment (robscott) [inline conformance/utils/kubernetes/helpers.go:187]: I'm not really sure what the solution is here, but this seems like something that could accidentally pass if a controller just hadn't gotten around to populating status yet. I hate to add delay for the sake of delay, but maybe we need to wait a few seconds before accepting `len(actual) == 0` as a successful result. If status is populated that seems sufficient to pass immediately, but not quite sure about the lack of conditions resulting in an immediate pass.

**Review by robscott — COMMENTED** [review]
Thanks @howardjohn! Mostly LGTM, just a couple nits.

File: conformance/utils/kubernetes/helpers.go (hunk 3 above)
Comment (howardjohn) [reply conformance/utils/kubernetes/helpers.go:187]: added a delay only in this case

**Pull request merged**